app/internal/page/app.go

```diff
@@ -535,6 +549,32 @@ func makeHandle(mux *http.ServeMux, logger Logger, store sessions.Store, localiz
 				}
 			}
 
+			if opt&RequireCertificateProviderSession != 0 {
```

Comment:
or do `RequireSession` and `CertificateProvider` so you'd set both (otherwise I need to put a check in to prevent both being set)